packages/espressocash_app/lib/features/kyc_sharing/utils/kyc_exception.dart

```diff
@@ -1,35 +1,109 @@
 import 'package:dio/dio.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
+import '../../../utils/errors.dart';
 
 part 'kyc_exception.freezed.dart';
 
 @freezed
 sealed class KycException with _$KycException implements Exception {
-  const factory KycException.invalidCode() = KycInvalidCode;
-  const factory KycException.invalidEmail() = KycInvalidEmail;
-  const factory KycException.invalidPhone() = KycInvalidPhone;
-  const factory KycException.invalidData() = KycInvalidData;
+  // Phone verification errors
+  const factory KycException.phone(PhoneError error) = KycPhoneError;
+
+  // Email verification errors
+  const factory KycException.email(EmailError error) = KycEmailError;
+
+  // Generic KYC errors
   const factory KycException.invalidToken() = KycInvalidToken;
+  const factory KycException.invalidData() = KycInvalidData;
   const factory KycException.genericError() = KycGenericError;
+  const factory KycException.invalid() = KycInvalid;
+  const factory KycException.invalidCode() = KycInvalidCode;
+}
+
+enum PhoneError {
+  blocked,
+  fraudBlock,
+  tooManyAttempts,
+  undeliverable,
+}
+
+enum EmailError {
+  blocked,
+  bounced,
+  spam,
 }
 
 extension ErrorExt on Exception {
   KycException toKycException() {
     if (this is! DioException) {
+      reportError(this, StackTrace.current);
+
       return const KycException.genericError();
     }
 
     final dioException = this as DioException;
     final message = (dioException.response?.data
         as Map<String, dynamic>?)?['message'] as String?;
+    final code =
+        (dioException.response?.data as Map<String, dynamic>?)?['code'] as int?;
+
+    if (message == null) {
+      reportError(dioException, StackTrace.current);
+
+      return const KycException.genericError();
+    }
+
+    // Phone-related Twilio errors
+    if (code == 2) {
+      final exception = switch (message) {
+        final s when s.contains('60203') =>
```

Comment:
That's not maintainable, let's think about proper error responding on backend side.